src/Params/ReportMakeParams.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Avtocod\B2BApi\Params;
+
+class ReportMakeParams
+{
+    /**
+     * Unique report type ID (e.g.: `some_report_uid` or `some_report_uid@domain`).
+     *
+     * @var string
+     */
+    private $report_type_uid;
+
+    /**
+     * Request type (e.g.: `VIN`, `GRZ`, `STS`, `PTS`, `CHASSIS`, etc.).
+     *
+     * @var string
+     */
+    private $type;
+
+    /**
+     * Request body (e.g.: `Z94CB41AAGR323020` (VIN), `А111АА177` (GRZ)).
+     *
+     * @var string
+     */
+    private $value;
+
+    /**
+     * Additional request options.
+     *
+     * @var array<mixed>|null
+     */
+    private $options = [];
+
+    /**
+     * Force update report, if it already was generated previously.
+     *
+     * @var bool
+     */
+    private $is_force = false;
+
+    /**
+     * URL to call (using `post` method) when report content updated.
+     *
+     * @var string|null
+     */
+    private $on_update_url;
+
+    /**
+     * URL to call (using `post` method) when report generation completed.
+     *
+     * @var string|null
+     */
+    private $on_complete_url;
+
+    /**
+     * Additional request data.
+     *
+     * @var array<mixed>|null
+     */
+    private $data;
+
+    /**
+     * Idempotence key which the server uses to recognize subsequent retries of the same request.
+     *
+     * @var string|null
+     */
+    private $idempotence_key;
+
+    /**
+     * ReportMakeParameter constructor.
+     *
+     * @param string $report_type_uid Unique report type ID (e.g.: `some_report_uid` or `some_report_uid@domain`)
+     * @param string $type            Request type (e.g.: `VIN`, `GRZ`, `STS`, `PTS`, `CHASSIS`, etc.)
+     * @param string $value           Request body (e.g.: `Z94CB41AAGR323020` (VIN), `А111АА177` (GRZ))
+     */
+    public function __construct(string $report_type_uid, string $type, string $value)
+    {
+        $this->report_type_uid = $report_type_uid;
+        $this->type            = $type;
+        $this->value           = $value;
+    }
+
+    /**
+     * @return string
+     */
+    public function getReportTypeUid(): string
+    {
+        return $this->report_type_uid;
+    }
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @return string
+     */
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    /**
+     * @param array<mixed>|null $options
+     *
+     * @return $this
+     */
+    public function setOptions(?array $options): self
```

Comment:
why options is optional?

Reply:
I was relying on the old method signature:

```php
public function userReportMake(string $report_type_uid,
                               string $type,	
                               string $value,	
                               ?array $options = [],	
                               ?bool $is_force = false,	
                               ?string $on_update = null,	
                               ?string $on_complete = null,	
                               ?array $data = null): UserReportMakeResponse
```

`?array $options = []` was optional.

Reply:
Do not take care about previous signatures. Here and in similar comments

Reply:
Ok, if we don't care about the old signature, there is no need to use `?` in the setters.

Fixed c0d4850d9eed2af9544b00c66e8bf26888e8b30a for:

- `options`
- `on_update_url`
- `on_complete_url`
- `data` 
- `idempotence_key`

Reply:
Of course!